### PR TITLE
Gbe 569: Fix registration reports for conference switching

### PR DESCRIPTION
**Pull request by jonkiparsky**

### CODE REVIEW

File: expo/gbe/report_views.py [hunk 1]
```diff
@@ -165,14 +181,30 @@ def env_stuff(request, conference_choice=None):
...
+    else:
+        conference = conf.Conference.current_conf()
+
     if profile_id == 'All':
         people = conf.Profile.objects.all().select_related()
     else:
-        people = []
+        people = []  #  Set it to be self, in list format
+
+    tmp_peeps = []
```
Comment (jonkiparsky) [inline expo/gbe/report_views.py]: These variable names are not acceptable. "people", "person" and "event" are the words we're looking for here.

File: expo/gbe/report_views.py [hunk 2]
```diff
@@ -165,14 +181,30 @@ def env_stuff(request, conference_choice=None):
...
+    for peep in range(0,len(people)):
+        for vent in range(0, len(people[peep].schedule)):
```
Comment (jonkiparsky) [inline expo/gbe/report_views.py]: why not 
```
for event in schedule:
```

File: expo/gbe/report_views.py [hunk 3]
```diff
@@ -165,14 +181,30 @@ def env_stuff(request, conference_choice=None):
...
+                tmp_peeps.append(people[peep])
+                break  #  Yes, I know this is bad form, refactor later
+    people = tmp_peeps
+    del tmp_peeps
 
     return render(request,
                   'gbe/report/printable_schedules.tmpl',
-                  {'people': people})
+                  {'people': people,
+                   'conference_slugs': conference_slugs,  'conference': conference})
```
Comment (jonkiparsky) [inline expo/gbe/report_views.py]: Please maintain dicts as either horizontal (leave them on one line) or vertical (separate each item on its own line). This is messy.

File: expo/gbe/report_views.py [hunk 4]
```diff
@@ -349,21 +394,38 @@ def room_schedule(request, room_id=None):
...
+        conference = conf.Conference.by_slug(request.GET['conf_slug'])
+    else:
+        conference = conf.Conference.current_conf()
+
+    conf_days = conference.conferenceday_set.all()
+    tmp_days = []
+    for position in range(0, len(conf_days)):
+        tmp_days.append(conf_days[position].day)
+    conf_days = tmp_days
+    del tmp_days
```
Comment (jonkiparsky) [inline expo/gbe/report_views.py]: Not needed, the variable goes out of scope immediately.

File: expo/gbe/report_views.py [hunk 5]
```diff
@@ -338,6 +376,13 @@ def room_schedule(request, room_id=None):
         except:
             rooms = []
 
+    conf_days = conference.conferenceday_set.all()
+    tmp_days = []
+    for position in range(0, len(conf_days)):
+        tmp_days.append(conf_days[position].day)
+    conf_days = tmp_days
+    del tmp_days
```
Comment (jonkiparsky) [inline expo/gbe/report_views.py]: Not needed, goes out of scope

File: expo/gbe/report_views.py [hunk 6]
```diff
@@ -165,14 +181,30 @@ def env_stuff(request, conference_choice=None):
...
+        people = []  #  Set it to be self, in list format
+
+    tmp_peeps = []
+    for peep in range(0,len(people)):
+        for vent in range(0, len(people[peep].schedule)):
+            if conference.conference_slug == people[peep].schedule[vent].eventitem.get_conference().conference_slug:
+                tmp_peeps.append(people[peep])
+                break  #  Yes, I know this is bad form, refactor later
+    people = tmp_peeps
+    del tmp_peeps
```
Comment (jonkiparsky) [inline expo/gbe/report_views.py]: Not needed, goes out of scope

File: expo/gbe/report_views.py [hunk 7]
```diff
@@ -165,14 +181,30 @@ def env_stuff(request, conference_choice=None):
...
+        conference = conf.Conference.current_conf()
+
     if profile_id == 'All':
         people = conf.Profile.objects.all().select_related()
     else:
-        people = []
+        people = []  #  Set it to be self, in list format
+
+    tmp_peeps = []
+    for peep in range(0,len(people)):
```
Comment (jonkiparsky) [inline expo/gbe/report_views.py]: If profile_id != "All", then len(people) = 0, and we don't do any of the following and just pass the empty list down into the context. What case does that break?

**Comment by jonkiparsky** [comment]
@BenignCremator I've done some cleanup, but there are some issues that should be addressed before this gets merged into master.

**Comment by BenignCremator** [comment]
On 01/19/2016 01:24 AM, Jon Kiparsky wrote:
> @BenignCremator <https://github.com/BenignCremator> I've done some
> cleanup, but there are some issues that should be addressed before this
> gets merged into master.

Ok, will take a look at it tonight.  Anything in particular?  The read 
through notes did not look like difficult or functional issues.

Hunter

**Comment by jonkiparsky** [comment]
Well, all of it is important enough that I brought it up. If they're not difficult, that's good. 
I agree that none of these items are "functional" in that they don't effect the behavior of the code, but getting this stuff wrong makes it a lot harder to work on the code in the future, so in that sense they're very functional.
(note - my style cleanup seems to have outdated some of my comments. Please be sure to address those as well)

**Comment by BenignCremator** [comment]
On 01/19/2016 05:15 PM, Jon Kiparsky wrote:
> Well, all of it is important enough that I brought it up. If they're not
> difficult, that's good.

That should take care of them.  Hopefully, we are now good to merge.

Hunter

**Comment by jonkiparsky** [comment]
@bethlakshmi Looks okay to me, but you should review to be sure the functionality is what you expect.

**Comment by jonkiparsky** [comment]
@bethlakshmi Feel free to merge this when you're satisfied.

**Comment by bethlakshmi** [comment]
OK.... I'm looking now and I'm really confused, because my integration manual tests aren't working.  I haven't dug into the code yet, but the user flow is simple enough and my data is from live, and realistic enough.. that I think there are serious issues....  I'm going to document failed manual testing flow by flow, once I have a clue what all is going on, I may start in trying to fix it.

**Comment by bethlakshmi** [comment]
First problem... http://localhost:8282/reports/staff_area

I'm selecting "GBE2016" from the conference selector at the top of the page
... Pushing Select Conference
... and I see ALL the shows in the system - last year's and this years.
... and exactly 1 staff area - Photography

OK... digging in, I see that the staff areas are getting switched (photography is the 1 staff area we have for 2016) and the shows are NOT getting switched.

I will write a separate ticket since it's not what I wanted, but it's not radically broken.

**Comment by bethlakshmi** [comment]
http://localhost:8282/reports/schedule/room

I go to this page....
... I select "GBE 2016" and press select conference
... it's a totally blank page with nothing on it.

My DB is a sync from GBE live from a week ago and there are many, many special events and classes with rooms booked, as you can see in the calendar pages (http://gbelive.webfactional.com/scheduler/Class/Saturday) so I'm not sure why it's not coming up...

Interestingly, GBE 2015 is working great.

This one is a blocker... we need the 2016 schedules most of all.

**Comment by bethlakshmi** [comment]
http://localhost:8282/reports/setup/room?conf_slug=GBE2015&submit=Select+conference

The switch part looks good... until I dig into the data, and find out that when the GBE 2015 switch is on, we are missing anything for Feb. 22. Many rooms on this list were scheduled for both days, (Sat and Sun), and we display schedules for both:

http://gbelive.webfactional.com/reports/setup/room

shows the data I'm working with, and I know we have the 22nd in there. I have trouble believing my data locally is different...

**Comment by bethlakshmi** [comment]
Getting close now... there's just one more:
http://localhost:8282/reports/schedule/all

Doesn't filter schedules per conference.
For example, Betty Blaize has her GBE 2015 events in there, even when you've selected 2016.

This one is annoying and fussy, and I'm headed in now to fix it.  The logic has to filter on two levels - 
1 - don't even make a schedule letter for people with no participation in the selected conference.
2 - make sure that the person only sees this year's items in their schedule.

**Comment by bethlakshmi** [comment]
I'm satisfied.
I'm crossing my fingers that none of this will be controversial - each fix was isolated to a single function, did not break (or add) any unit tests (feel free to hit me with tickets for more unit tests on these updates...), and I pep8 cleaned up my own tracks.

**Pull request merged**